app/services/api/presenters/event.rb

```diff
@@ -0,0 +1,56 @@
+module Api
+  module Presenters
+    class Event
+      attr_reader :event
+
+      def self.call(event)
+        new(event).call
+      end
+
+      def initialize(event)
+        @event = event
+      end
+
+      def call
+        {
+          id: event.id,
+          eventable_id: event.eventable_id,
+          eventable_type: event.eventable_type,
+          title: event.title,
+          event_type: event.event_type,
+          date: event.date,
+          description: event.description,
+          eventable_title: event.eventable_title,
+          link: link(event.eventable_type, event.eventable_id)
+        }
+      end
+
+      private
+
+      def link(eventable_type, eventable_id)
+        if eventable_type.eql?('Legislation')
+          legislation_route(Legislation.find(eventable_id)&.geography_id, Legislation.find(eventable_id))
+        elsif eventable_type.eql?('Litigation')
+          url.cclow_geography_litigation_case_path(
+            Litigation.find(eventable_id)&.geography_id,
```

Comment:
Do we need to pass the eventable_type and eventable_id separately? otherwise we could just have the `eventable` object and from there get to `geography_id`.
Example: if you go to the rails console and get the first event: `e = Event.first` and then access `e.eventable` that will be the correct object: Litigation/Legislation/Target/Geography. So we can make use of those associations, instead of using the find here.

Reply:
true, we don't need to pass anything to these internal methods. I pushed the commit with a simplified code